ratelimit.go

```diff
@@ -89,6 +89,13 @@ func withoutSlackOption(l *limiter) {
 	l.maxSlack = 0
 }
 
+// WithCustomDuration is an option to set durations not based around seconds.
+func WithCustomDuration(amount int, duration time.Duration) Option {
```

Comment:
Oh, this is unfortunate, but clearly necessary. Can we name this `WithCountPerDuration` and clarify in the comment that this overrides the `rate` argument of the constructor?